### ISSUE TITLE
generate-website crashes as soon as a deployment diagram is added

### ISSUE THREAD

**Issue by YoeriVD** [avisi-cloud/structurizr-site-generatr#151]
The generator has been doing a wonderful job so far. 

However, today I added a deployment diagram (nothing fancy):

```c4
deployment mySystem "Production" "ProductionDeployment" {
    include *
    autoLayout
}
```

Which results in the following stacktrace:
```
ProductionDeployment-ProductionDeployment UP-TO-DATE
Exception in thread "main" java.io.FileNotFoundException
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:484)
        at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:542)
        at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:567)
        at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:670)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:765)
        at nl.avisi.structurizr.site.generatr.site.SiteGeneratorKt.generateHtmlFiles(SiteGenerator.kt:134)
        at nl.avisi.structurizr.site.generatr.site.SiteGeneratorKt.generateSite(SiteGenerator.kt:64)
        at nl.avisi.structurizr.site.generatr.site.SiteGeneratorKt.generateSite$default(SiteGenerator.kt:47)
        at nl.avisi.structurizr.site.generatr.GenerateSiteCommand.generateSiteForModel(GenerateSiteCommand.kt:98)
        at nl.avisi.structurizr.site.generatr.GenerateSiteCommand.execute(GenerateSiteCommand.kt:67)
        at kotlinx.cli.ArgParser.parse(ArgParser.kt:657)
        at kotlinx.cli.ArgParser.parse(ArgParser.kt:530)
        at nl.avisi.structurizr.site.generatr.AppKt.main(App.kt:13)
Caused by: java.io.FileNotFoundException: dist/svg/ProductionDeployment-ProductionDeployment.svg (No such file or directory)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:158)
        at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:125)
        at kotlin.io.FilesKt__FileReadWriteKt.readText$default(FileReadWrite.kt:125)
        at nl.avisi.structurizr.site.generatr.site.DiagramGeneratorKt.readSvg(DiagramGenerator.kt:85)
        at nl.avisi.structurizr.site.generatr.site.DiagramGeneratorKt.generateDiagramWithElementLinks(DiagramGenerator.kt:52)
        at nl.avisi.structurizr.site.generatr.site.SiteGeneratorKt$generateSite$generatorContext$1.invoke(SiteGenerator.kt:58)
        at nl.avisi.structurizr.site.generatr.site.SiteGeneratorKt$generateSite$generatorContext$1.invoke(SiteGenerator.kt:56)
        at nl.avisi.structurizr.site.generatr.site.model.DiagramViewModel$Companion.forView(DiagramViewModel.kt:23)
        at nl.avisi.structurizr.site.generatr.site.model.MarkdownToHtmlKt.transformEmbeddedDiagramElements(MarkdownToHtml.kt:75)
        at nl.avisi.structurizr.site.generatr.site.model.MarkdownToHtmlKt.markdownToHtml(MarkdownToHtml.kt:33)
        at nl.avisi.structurizr.site.generatr.site.model.WorkspaceDocumentationSectionPageViewModel.<init>(WorkspaceDocumentationSectionPageViewModel.kt:12)
        at nl.avisi.structurizr.site.generatr.site.SiteGeneratorKt$generateHtmlFiles$1$5$1.invoke(SiteGenerator.kt:107)
        at nl.avisi.structurizr.site.generatr.site.SiteGeneratorKt$generateHtmlFiles$1$5$1.invoke(SiteGenerator.kt:107)
        at nl.avisi.structurizr.site.generatr.site.SiteGeneratorKt$generateHtmlFiles$2.invoke(SiteGenerator.kt:134)
        at nl.avisi.structurizr.site.generatr.site.SiteGeneratorKt$generateHtmlFiles$2.invoke(SiteGenerator.kt:134)
        at nl.avisi.structurizr.site.generatr.site.SiteGeneratorKt.generateHtmlFiles$lambda$14(SiteGenerator.kt:134)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:722)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
        at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1311)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1841)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1806)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

The same workspace works perfectly well in structurizr lite.

Any thoughts or ideas? :)

**Comment by YoeriVD** [comment]
Tried it again and after the nth time it suddenly started working on github and locally.
I think it might have been an older java jdk issue.

**Comment by YoeriVD** [comment]
I have found the difference!

It works when I run the generator without the deployment view first. (comment out the part in views, not the model)

1. comment deployment diagram
2. generate site
3. re-enable deployment diagram
4. generate site
5. works

as soon as I delete the cached build folder and try it again, I run into the same issue and the steps above solve it every time.
It is however a problem in CI :/

**Comment by simonbrowndotje** [comment]
Another data point ... I see similar crashes with website generation and website serving on a "random" basis. I've been showing Site Generatr in some conference talks recently, and it will sometimes fail during the talk after working just beforehand, so I also suspect it's something related to the build cache.

**Comment by YoeriVD** [comment]
> Hi @YoeriVD thanks for the report. Could be please add a minimum reproduction model or a diff to the example model so that this issue appears?

I'd rather not share it publicly, but I am willing to give you temporary read access to our repository so you can inspect the logs of the failed actions.
Would that help?

**Comment by jp7677** [comment]
@YoeriVD Thanks for the perfect reproduction scenario!
PR https://github.com/avisi-cloud/structurizr-site-generatr/pull/154 should address this. It was indeed an error related to checking if diagram had already been generated or. Since all pages are generated concurrently, this also explains why it still sometimes succeeded with the same model on initial generation.
 
Feel to give it a go and comment on the PR.

**Comment by YoeriVD** [comment]
> @YoeriVD Thanks for the perfect reproduction scenario! PR #154 should address this. It was indeed an error related to checking if a diagram had already been generated or not. Since all pages are generated concurrently, this also explains why it still sometimes succeeded with the same model on initial generation.
> 
> Feel to give it a go and comment on the PR.

Works!

**Comment by dirkgroot** [comment]
> Another data point ... I see similar crashes with website generation and website serving on a "random" basis. I've been showing Site Generatr in some conference talks recently, and it will sometimes fail during the talk after working just beforehand, so I also suspect it's something related to the build cache.

Cool to read you're showing this off in your talks, @simonbrowndotje! #154 has been merged, and a new version (1.0.24) has been released. Could you reopen this issue if you experience any crashes with this new version?
